combiner.go

```diff
@@ -0,0 +1,55 @@
+package krakend
+
+import (
+	"net/http"
+	"github.com/devopsfaith/krakend/proxy"
+)
+
+func combineStatusAndData(total int, parts []*proxy.Response) *proxy.Response {
+	isComplete := len(parts) == total
+	var retResponse *proxy.Response
+	var retStatus int
+
+	for _, part := range parts {
+		if part == nil || part.Data == nil {
+			isComplete = false
+			continue
+		}
+
+		isComplete = isComplete && part.IsComplete
+		if retResponse == nil {
+			retResponse = part
+			continue
+		}
+
+		status := part.Metadata.StatusCode
+		if status != 0 { // failed
+			if retStatus == 0 {
+				retStatus = status
+			} else if retStatus != status {
+				retStatus = http.StatusInternalServerError
+			}
+		}
+
+		for k, v := range part.Data {
+			retResponse.Data[k] = v
+		}
+	}
+
+	if retStatus > http.StatusInternalServerError {
+		retStatus = http.StatusInternalServerError
```

Comment:
`if isComplete == false` maybe also ? There may be one path of code where `resStatus` is still 0 here

Reply:
Ha that might be covered in `jsonErrorRender` :bowing_man:

Reply:
@PuKoren I think this is something we should seriously consider though, the krakend responds with status 200 and with a possibly incomplete body when at least one of the backend hosts succeeded, which is different from what we discussed.
Didn't think about this well...

Reply:
Will keep this in mind, for when it causes some problem.